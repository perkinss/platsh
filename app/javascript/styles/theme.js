import {createMuiTheme} from "@material-ui/core/styles";

const theme = createMuiTheme({
    palette: {
        primary: {
            main: '#512da8',
        },
        secondary: {
            main: '#ab47bc',
        },
        type: 'dark'
    },
    slider: {
        selectionColor: '#F76B1C',
        handleFillColor: '#F76B1C',
        thumb: 'secondary'
    }

})

export default theme
