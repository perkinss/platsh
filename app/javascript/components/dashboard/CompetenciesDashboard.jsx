import React from 'react'
import {Typography, Table, TableCell, TableRow, TableFooter, TablePagination, TableSortLabel, TableBody, TableHead,
    Icon, Chip, Avatar, LinearProgress, Grow, Tooltip, Zoom} from '@material-ui/core'
import Star from '@material-ui/icons/StarBorderRounded'
import { makeStyles } from "@material-ui/styles/index";

const useStyles = makeStyles(theme => ({
    progressBar: {
        minHeight: '28px',
        backgroundColor: theme.palette.grey[800],
        opacity: 1,
    },
}))

const standardBarStyle = makeStyles(theme => ({
    bar: {
        backgroundColor: 'rgba(0, 227, 150, 0.85)',
        borderRadius: '2px'
    }
}))

const topBarStyle = makeStyles(theme => ({
    bar: {
        backgroundColor: theme.palette.primary.light,
        borderRadius: '2px'
    }
}))

const tooltipStyles = makeStyles(theme => ({
    tooltip: {
        fontSize: '18px',
        marginLeft: '-120px',
        width: '70px',
        textAlign: 'center'
    }
}))


export default function CompetenciesDashboard(props) {
    const { show, details, theRef } = props
    const progressClasses = useStyles();
    const topBarClasses = topBarStyle();
    const tooltipClasses = tooltipStyles();
    const standardBarClasses = standardBarStyle()

    if (!show || !details.details) {
        return null
    }
    const classModeAverage = details.details.reduce((sum, detail) => {
        return sum + detail.mode
    }, 0)/details.details.length

    const calculateModePercent = (value) => {
         return (value / 4.0 * 100)
    }

    const classPercentage = calculateModePercent(classModeAverage)

    return (
        <React.Fragment>
            <Typography paragraph variant={'h6'} ref={theRef}>{details.name||details.title}</Typography>
            <div >
            <Table size={'small'} autoFocus>
                <TableHead>
                    <TableRow>
                        <TableCell rowSpan={2}>Name</TableCell>
                        <TableCell rowSpan={2}>
                            Number of tasks
                        </TableCell>
                        <TableCell width={'50%'}>
                            Class Averages
                        </TableCell>
                    </TableRow>
                    <TableRow>
                        <TableCell>
                            <Tooltip classes={tooltipClasses} title={classModeAverage.toFixed(2)} placement="right-start" TransitionComponent={Zoom}>
                                <LinearProgress
                                    classes={topBarClasses}
                                    className={progressClasses.progressBar}
                                    variant='determinate' value={classPercentage}>
                                </LinearProgress>
                            </Tooltip>

                        </TableCell>
                    </TableRow>
                </TableHead>
                <TableBody >
                    {details.details.map((detail) => {
                        return (
                            <TableRow key={`row-${detail.description}`}>
                                <TableCell  >
                                    {detail.description}
                                </TableCell>
                                <TableCell  >
                                    {detail.count}
                                </TableCell>
                                <TableCell width={'70%'}>
                                    {detail && detail.mode && <Grow
                                        in={show}
                                        style={{ transformOrigin: '0 0 0' }}
                                        timeout={1000}
                                    >
                                        <Tooltip classes={tooltipClasses} title={detail.mode}
                                                 placement="right-start" TransitionComponent={Zoom}>
                                            <LinearProgress
                                                color={'primary'}
                                                classes={standardBarClasses}
                                                className={progressClasses.progressBar}
                                                variant='determinate'
                                                value={calculateModePercent(detail.mode)}/>
                                        </Tooltip>
                                    </Grow>}
                                </TableCell>
                            </TableRow>
                        )
                    })}

                </TableBody>
            </Table>
            </div>
        </React.Fragment>
    )

}