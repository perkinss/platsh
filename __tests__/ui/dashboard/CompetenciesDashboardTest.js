import React from 'react'
import { createRender } from '@material-ui/core/test-utils';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect'
import theme from "../../../app/javascript/styles/theme";
import { ThemeProvider } from '@material-ui/styles'
import CompetenciesDashboard from "../../../app/javascript/components/dashboard/CompetenciesDashboard";

Enzyme.configure({ adapter: new Adapter() });

describe('<CompetenciesDashboard />', () => {
    let render;
    let classes = {
        content: {}
    }

    let section = {
        name: "New at the Zoo...",
        competency_stats: [
            {title: "Bee kind",
                id: 1,
                percent_included: "71",
                percent_observed: "22",
                details: [
                    {
                        mode: 3,
                        description: "Bee kind to bees every day",
                        count: 23,
                        id: 8,
                    }]
            },
            {
                title: "Communicating and representing",
                id: 2,
                percent_included: "0",
                percent_observed: "0",
                details: [
                    {
                        mode: 0,
                        description: "what we have heeere is a failure to commun'cate",
                        count: 0,
                        id: 3,
                    }]
                    },
        ]
    }
    beforeAll(() => {
        render = createRender();
    })

    it('should match the snapshot', () => {
        const wrapper = render(
            <ThemeProvider theme={theme}>
                    <CompetenciesDashboard classes={classes} section={section} />
            </ThemeProvider>
                );
        expect(wrapper).not.toBeNull()
        expect(wrapper).toMatchSnapshot()
    });

    it('should have the details', () => {
        const wrapper = render(
            <ThemeProvider theme={theme}>
                <CompetenciesDashboard classes={classes} show={true} details={section.competency_stats[0]} />
            </ThemeProvider>
        );
        let result = wrapper.html()
        expect(result).toContain("Bee kind")
        expect(result).not.toContain("Communicating and representing")
    });

    it('should remain hidden if not showing', () => {
        const wrapper = render(
            <ThemeProvider theme={theme}>
                <CompetenciesDashboard classes={classes} show={false} details={section.competency_stats[1]} />
            </ThemeProvider>
        );
        let result = wrapper.html()
        expect(result).toBeNull()
    })

    it('should remain hidden if no details', () => {
        const wrapper = render(
            <ThemeProvider theme={theme}>
                <CompetenciesDashboard classes={classes} show={true} details={''}/>
            </ThemeProvider>
        );
        let result = wrapper.html()
        expect(result).toBeNull()
    })

})