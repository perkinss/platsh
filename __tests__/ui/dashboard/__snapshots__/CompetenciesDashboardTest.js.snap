// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompetenciesDashboard /> should match the snapshot 1`] = `
Object {
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
